profiles/docker/Dockerfile.tpl

```diff
@@ -1,103 +1,84 @@
-    apt-get install nodejs -y
-
-RUN curl -L -o apache-ant-${ANT_VERSION}-bin.tar.gz https://downloads.apache.org/ant/binaries/apache-ant-${ANT_VERSION}-bin.tar.gz \
-    && mkdir ant-${ANT_VERSION} \
-    && tar -zxvf apache-ant-${ANT_VERSION}-bin.tar.gz \
-    && mv apache-ant-${ANT_VERSION} ${ANT_HOME} \
-    && rm apache-ant-${ANT_VERSION}-bin.tar.gz \
-    && rm -rf ant-${ANT_VERSION} \
-    && rm -rf ${ANT_HOME}/manual \
-    && unset ANT_VERSION
-
-ENV PATH ${PATH}:${ANT_HOME}/bin
+ARG EXIST_VERSION=[[ $docker?eXist ]]
+ARG BUILD=local
 
-FROM builder as tei
+# START STAGE 1
+FROM ghcr.io/eeditiones/builder:latest AS builder
 
-ARG EXIST_VERSION=[[ $docker?eXist ]]
-ARG PUBLISHER_LIB_VERSION=[[ $docker?tei-publisher-lib ]]
 ARG ROUTER_VERSION=[[ $docker?roaster ]]
 ARG JWT_VERSION=[[ $docker?jwt ]]
 ARG CRYPTO_VERSION=[[ $docker?crypto ]]
 
-# add key
-RUN  mkdir -p ~/.ssh && ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+WORKDIR /tmp
+
+# Build jinks-template
+RUN git clone https://github.com/eeditiones/jinks-templates.git \
+    && cd jinks-templates \
+    && ant
 
 # Build tei-publisher-lib
-RUN if [ "${PUBLISHER_LIB_VERSION}" = "master" ]; then \
-        git clone https://github.com/eeditiones/tei-publisher-lib.git \
-        && cd tei-publisher-lib \
-        && ant \
-        && cp build/*.xar /tmp; \
-    else \
-        curl -L -o /tmp/tei-publisher-lib-${PUBLISHER_LIB_VERSION}.xar https://exist-db.org/exist/apps/public-repo/public/tei-publisher-lib-${PUBLISHER_LIB_VERSION}.xar; \
-    fi
+RUN git clone https://github.com/eeditiones/tei-publisher-lib.git \
+    && cd tei-publisher-lib \
+    && ant 
 
 # Build tei-publisher-app
 COPY . [[$pkg?abbrev]]/
 RUN  cd  [[$pkg?abbrev]] \
     && sed -i 's/$config:webcomponents :=.*;/$config:webcomponents := "local";/' modules/config.xqm \
     && ant xar-local
 
-RUN curl -L -o /tmp/roaster-${ROUTER_VERSION}.xar http://exist-db.org/exist/apps/public-repo/public/roaster-${ROUTER_VERSION}.xar
-RUN curl -L -o /tmp/jwt-${JWT_VERSION}.xar http://exist-db.org/exist/apps/public-repo/public/jwt-${JWT_VERSION}.xar
-RUN curl -L -o /usr/local/exist/autodeploy/expath-crypto-module-${CRYPTO_VERSION}.xar https://exist-db.org/exist/apps/public-repo/public/expath-crypto-module-${CRYPTO_VERSION}.xar
+ADD http://exist-db.org/exist/apps/public-repo/public/roaster-${ROUTER_VERSION}.xar 001.xar
+ADD http://exist-db.org/exist/apps/public-repo/public/jwt-${JWT_VERSION}.xar 002.xar
+ADD https://exist-db.org/exist/apps/public-repo/public/expath-crypto-module-${CRYPTO_VERSION}.xar 003.xar
+
+FROM duncdrum/existdb:${EXIST_VERSION} AS build_local
+
+ARG USR=root
+USER ${USR}
+
+ONBUILD COPY --from=builder /tmp/*.xar /exist/autodeploy/
+ONBUILD COPY --from=builder /tmp/jinks-templates/build/*.xar /exist/autodeploy/004.xar
+ONBUILD COPY --from=builder /tmp/tei-publisher-lib/build/*.xar /exist/autodeploy/005.xar
+ONBUILD COPY --from=builder /tmp/[[$pkg?abbrev]]/build/*.xar /exist/autodeploy/006.xar
```

Comment:
I'd recommend adding a config option for the data repo, usually the normal repo location with a `-data` postfix. Needed almost always so making it configurable would remove a big reason to divert from the vanilla tuttle output.

Reply:
I like the idea is now #90